applications/crossbar/src/modules/cb_auth.erl

```diff
@@ -113,20 +138,33 @@ authorize(Context, PathToken) ->
 
 -spec authorize(cb_context:context(), path_token(), path_token()) -> boolean().
 authorize(Context, PathToken, Id) ->
-    authorize_nouns(Context, PathToken, Id, cb_context:req_verb(Context), cb_context:req_nouns(Context)).
+    authorize_nouns_id(Context, PathToken, Id, cb_context:req_verb(Context), cb_context:req_nouns(Context)).
 
 -spec authorize_nouns(cb_context:context(), path_token(), req_verb(), req_nouns()) -> boolean().
-authorize_nouns(_Context, ?CALLBACK_PATH, ?HTTP_PUT, [{<<"auth">>, _}]) -> 'true';
-authorize_nouns(_Context, ?AUTHORIZE_PATH, ?HTTP_PUT, [{<<"auth">>, _}]) -> 'true';
-authorize_nouns(_Context, ?TOKENINFO_PATH, ?HTTP_GET, [{<<"auth">>, _}]) -> 'true';
-authorize_nouns(_Context, ?TOKENINFO_PATH, ?HTTP_POST, [{<<"auth">>, _}]) -> 'true';
+authorize_nouns(_, ?APPS_PATH,            ?HTTP_GET,   [{<<"auth">>, _}]) -> 'true';
+authorize_nouns(_, ?AUTHORIZE_PATH,       ?HTTP_PUT,   [{<<"auth">>, _}]) -> 'true';
+authorize_nouns(_, ?CALLBACK_PATH,        ?HTTP_PUT,   [{<<"auth">>, _}]) -> 'true';
+authorize_nouns(C, ?IDENTITY_SECRET_PATH, ?HTTP_PATCH, [{<<"auth">>, _}]) -> cb_context:is_superduper_admin(C);
+authorize_nouns(C, ?IDENTITY_SECRET_PATH, ?HTTP_PATCH, [{<<"auth">>, _}, {<<"accounts">>, _AccountId}]) -> cb_context:is_account_admin(C);
+authorize_nouns(_, ?LINKS_PATH,           ?HTTP_GET,   [{<<"auth">>, _}]) -> 'true';
+authorize_nouns(_, ?PROVIDERS_PATH,       ?HTTP_GET,   [{<<"auth">>, _}]) -> 'true';
+authorize_nouns(_, ?TOKENINFO_PATH,       ?HTTP_GET,   [{<<"auth">>, _}]) -> 'true';
+authorize_nouns(_, ?TOKENINFO_PATH,       ?HTTP_POST,  [{<<"auth">>, _}]) -> 'true';
 authorize_nouns(_, _, _, _) -> 'false'.
 
--spec authorize_nouns(cb_context:context(), path_token(), path_token(), req_verb(), req_nouns()) -> boolean().
-authorize_nouns(Context, ?LINKS_PATH, _Id, ?HTTP_PUT, [{<<"auth">>, _}]) ->
-    cb_context:is_authenticated(Context);
-authorize_nouns(Context, ?LINKS_PATH, _Id, ?HTTP_DELETE, [{<<"auth">>, _}]) ->
-    cb_context:is_authenticated(Context).
+-spec authorize_nouns_id(cb_context:context(), path_token(), path_token(), req_verb(), req_nouns()) -> boolean().
+authorize_nouns_id(_, ?APPS_PATH,      _Id,           ?HTTP_GET,    [{<<"auth">>, _}]) -> 'true';
+authorize_nouns_id(C, ?KEYS_PATH,      ?PRIVATE_PATH, ?HTTP_PATCH,  [{<<"auth">>, _}]) -> cb_context:is_superduper_admin(C);
+authorize_nouns_id(C, ?KEYS_PATH,      ?PUBLIC_PATH,  ?HTTP_GET,    [{<<"auth">>, _}]) -> cb_context:is_account_admin(C);
+authorize_nouns_id(_, ?LINKS_PATH,     _Id,           ?HTTP_GET,    [{<<"auth">>, _}]) -> 'true';
+%% monster-ui still uses this (accounts/123/auth/links)
+authorize_nouns_id(_, ?LINKS_PATH,     _Id,           ?HTTP_GET,    [{<<"auth">>, _}, [<<"accounts">>, _]]) -> 'true';
```

Comment:
I think it should be `[{auth, _}, {accounts, _}]`, you have square brackets around accounts